### PR TITLE
feat: added ComfyUI

### LINKED ISSUE #274: Add ComfyUI as a backend to run Stable Diffusion
### User Story

**Title**: Add ComfyUI as a Backend to Run Stable Diffusion  
**As a** Captain user  
**I want** to use ComfyUI as a backend for running Stable Diffusion within Captain  
**So that** I can leverage the ComfyUI interface for creating and managing Stable Diffusion workflows

### Acceptance Criteria

- [ ] ComfyUI is integrated into Captain as a backend option for running Stable Diffusion.
- [ ] The integration supports loading and running ComfyUI workflows directly within Captain.

### PR DESCRIPTION
**Pull request by TimPietrusky**
## Motivation

- Added ComfyUI as a backend to use Stable Diffusion

## Issues closed

closes #274

### CODE REVIEW

File: src/electron/helpers/services/comfyui.ts [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+			"--output-directory",
+			outputPath,
+		];
+
+		// Start ComfyUI process
+		this.process = execa(pythonBinaryPath, ["-u", scriptPath, ...scriptArguments]);
+
+		// Listen for stdout and stderr
+		if (this.process.stdout && this.process.stderr) {
+			this.process.stdout.on("data", async data => {
```
Comment (pixelass) [inline src/electron/helpers/services/comfyui.ts]: unneccessary async

File: src/electron/helpers/services/comfyui.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+				logger.info(data.toString());
+			});
+
+			this.process.stderr.on("data", data => {
+				logger.info(data.toString());
```
Comment (pixelass) [inline src/electron/helpers/services/comfyui.ts:179]: which type can data be? make sure we don't try to cast strings from objects.

File: src/electron/helpers/services/comfyui.ts [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+	/**
+	 * Sets up a WebSocket connection for a given client ID and attaches message handling.
+	 * It manages the lifecycle events of the WebSocket connection including opening, messages, closing, and errors.
+	 *
+	 * @param {string} clientId - The unique identifier for the client.
+	 * @param {WebSocketMessageHandler} handleMessage - The function to call when a message is received on the WebSocket.
+	 * @private
+	 */
+	private setupWebSocket(clientId: string, handleMessage: WebSocketMessageHandler) {
+		const ws = new WebSocket(`ws://${this.url}/ws?clientId=${clientId}`);
```
Comment (pixelass) [inline src/electron/helpers/services/comfyui.ts]: declarative names: `const webSocket = new WebSocket`

File: src/electron/helpers/services/comfyui.ts [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+	/**
+	 * Queues a new workflow for processing by the ComfyUI server.
+	 *
+	 * @param {object} prompt - The workflow data to be processed.
+	 * @param {string} clientId - The client ID associated with this request.
+	 * @returns {Promise<QueueWorkflowResponse>} The response from the server including prompt ID and any node errors.
+	 * @throws {Error} Throws an error if the request to queue the workflow fails.
+	 */
+	public async queueWorkflow(prompt: object, clientId: string): Promise<QueueWorkflowResponse> {
```
Comment (pixelass) [inline src/electron/helpers/services/comfyui.ts]: we could use either `Record<string, unknown>` or the type we use in Broom

```ts
interface Node<T extends Record<string, unknown> = Record<string, unknown>> {
	inputs: T;
	class_type: string;
}


// Represents the node chain as a mapping from string to Node
export type NodeChain = Record<string, Node>;
```

**Review by pixelass — CHANGES_REQUESTED** [review]

File: src/electron/helpers/services/comfyui.ts (hunk 1 above)
Comment (TimPietrusky) [reply src/electron/helpers/services/comfyui.ts]: This was resolved.

File: src/electron/helpers/services/comfyui.ts (hunk 2 above)
Comment (TimPietrusky) [reply src/electron/helpers/services/comfyui.ts:179]: I added a type.

File: src/electron/helpers/services/comfyui.ts (hunk 4 above)
Comment (TimPietrusky) [reply src/electron/helpers/services/comfyui.ts]: Added the type as suggested.

**Review by pixelass — APPROVED** [review]

**Pull request merged**